src/spartan/upsnark.rs

```diff
@@ -178,18 +184,31 @@ impl<G: Group, EE: EvaluationEngineTrait<G>> RelaxedR1CSSNARKTrait<G> for R1CSSN
     let _ = circuit.synthesize(&mut cs);
 
     // Commits to witness (expensive)
+    let hollow_S = R1CSShape::<G> {
+      num_cons: pk.num_cons_total,
+      num_vars: pk.num_vars_total,
+      num_io: 0,
+      A: vec![],
+      B: vec![],
+      C: vec![],
+    };
     let (u, w) = cs
-      .r1cs_instance_and_witness(&pk.S, &pk.ck)
+      // .r1cs_instance_and_witness(&pk.S, &pk.ck)
+      .r1cs_instance_and_witness(&hollow_S, &pk.ck)
       .map_err(|_e| SpartanError::UnSat)?;
 
     let non_commitment_span = tracing::span!(tracing::Level::INFO, "PostCommitProve");
     let _guard = non_commitment_span.enter();
 
-    let W = w.pad(&pk.S); // pad the witness
+    // form a padded version of the witness, W, for use in polynomial commitment later 
+    let mut W = w.clone(); 
+    W.W.extend(vec![G::Scalar::ZERO; pk.num_vars_total - W.W.len()]);
+    // let W = w.pad(&pk.S); // pad the witness
```

Comment:
nit: rm

Reply:
I think this entire cloning is unnecessary because it's already pre-padded. I've pushed an update with it removed. (Tests pass)